### DISCUSSION TITLE
Plugin: Smart workspace switcher

### DISCUSSION THREAD

**Discussion by MLFlexer** [wezterm/wezterm#4040] [Show and tell]
# Introducing smart_workspace_switcher.wezterm

A smart wezterm workspace switcher inspired by [joshmedeski/t-smart-tmux-session-manager](https://github.com/joshmedeski/t-smart-tmux-session-manager)
## Usage
💨 Level up your workflow by switching between workspaces ⚡ ***BLAZINGLY FAST*** ⚡ with 1️⃣ keypress, the power of fuzzy finding and zoxide! 💨

[Demo video](https://youtu.be/AhmSPRC6Uc4)

## Link

[Plugin link](https://github.com/MLFlexer/smart_workspace_switcher.wezterm)

**Comment by wez** [comment]
Thanks for sharing this!

**Comment by Horryportier** [comment]
I just was about to try make something like this but i see i don't have to thanks

**Reply by MLFlexer** [reply]
Saved you a bit of work then. If you have any improvements, feel free to make a PR or an issue!

**Reply by Horryportier** [reply]
sure for know i'm confused with PATH_TO_SCRIPT

**Reply by MLFlexer** [reply]
If you are trying to configure it manually, then you must save the shell script somewhere, and then reference it, by replacing the path

**Comment by qujihan** [comment]
Can you share your tab bar configuration?

**Reply by MLFlexer** [reply]
Sure, here you go: [https://github.com/MLFlexer/.dotfiles/blob/main/wezterm/bar.lua](https://github.com/MLFlexer/.dotfiles/blob/main/wezterm/bar.lua)

**Comment by johannhmnn** [comment]
I'm not sure why, but by setting up the way it is instructed in the repo I can't seem to activate via keybindings. When I do it, it sort of only blinks. I'll attach a video showing the problem. I've tried the script alone and it works, and tried `window:spawn` with other programs and it also worked (though I still got some errors about the PATH variable, I'm not sure how it works). The debug overlay does not show any errors when this happens. I'm using fish shell, but also tried it while setting the default shell to zsh.


https://github.com/wez/wezterm/assets/62679700/b3cc4307-6e8c-4f38-80b2-1fa92604b6ba

**Reply by MLFlexer** [reply]
have you tried replacing the args with:

```lua
args = {
'bin/bash',
'/Users/johannhomonnai/.config/wezterm/scripts/workspace_switcher.sh',
'--tab-id',
tostring(current_tab_id)
}
```

If that does not work, can you try and replace `tab:set_title(wezterm.nerdfonts.md_dock_window .. 'Workspace Switcher')` with `tab:set_title(wezterm.nerdfonts.mdi_dock_window .. 'Workspace Switcher')`

**Reply by johannhmnn** [reply]
Unfortunately it still presents the same behavior

https://github.com/wez/wezterm/assets/62679700/287347e1-4fab-455e-9854-488353e5d42b

**Reply by johannhmnn** [reply]
I mean, I don't really really bother because I can still use the alias `ws` to use the script, it's just I have no idea what is happening. Congrats on the plugin, by the way!

**Reply by MLFlexer** [reply]
Sorry I was missing a backslash in the args command, try to change `'bin/bash',` to `'/bin/bash',`

```lua
args = {
'/bin/bash',
'/Users/johannhomonnai/.config/wezterm/scripts/workspace_switcher.sh',
'--tab-id',
tostring(current_tab_id)
}
```

or reference bash in some other way

**Reply by johannhmnn** [reply]
Still nothing 😅 I really have no clue

**Reply by MLFlexer** [reply]
Can you try and see if `/bin/bash` is the correct path to bash, by typing `which bash`, you can also try and remove the bash part:
```lua
args = {
'/Users/johannhomonnai/.config/wezterm/scripts/workspace_switcher.sh',
'--tab-id',
tostring(current_tab_id)
}
```

**Reply by johannhmnn** [reply]
It didn't work even like that. But thanks anyway!

**Reply by johannhmnn** [reply]
Hi, just an update: turns out I had to set the PATH using `config.environment_variables`. Now it works!

**Reply by MLFlexer** [reply]
Awesome, glad it works!
